src/isomorphic/classic/class/ReactClass.js

```diff
@@ -728,6 +728,11 @@ var ReactClassMixin = {
    * type signature and the only use case for this, is to avoid that.
    */
   replaceState: function(newState, callback) {
+    if (this.updater.isFiberUpdater) {
```

Comment:
There was a bug because `enqueueUpdate` only adding the callback to the queue; it didn't schedule an update on the fiber. This was unobservable previously because the callback was added to the same update queue node as the state update.

I can get it to work by adding that to `enqueueUpdate`, but it felt wrong that you need an extra flush for a callback that is conceptually linked to the update.